tests/testing-framework/src/spin.rs

```diff
@@ -10,17 +12,34 @@ pub struct Spin {
     #[allow(dead_code)]
     stdout: OutputStream,
     stderr: OutputStream,
-    port: u16,
+    io_mode: IoMode,
 }
 
 impl Spin {
+    pub fn start(
+        spin_binary_path: &Path,
+        current_dir: &Path,
+        spin_up_args: Vec<impl AsRef<std::ffi::OsStr>>,
+        mode: SpinMode,
+    ) -> anyhow::Result<Self> {
+        match mode {
+            SpinMode::Http => Self::start_http(spin_binary_path, current_dir, spin_up_args),
```

Comment:
I've heard some chat about support multiple types of trigger at some stage. would that become a different SpinMode at that time?

Reply:
Yes. The `SpinMode` is used to indicate how we know how to communicate with Spin.